octobot_trading/data/order.py

```
@@ -254,38 +264,67 @@ def is_self_managed(self):
+            "filled_price": get_value_or_default(raw_order, ExchangeConstantsOrderColumns.PRICE.value, None),
+            "total_cost": get_value_or_default(raw_order, ExchangeConstantsOrderColumns.COST.value, None),
+            "fee": get_value_or_default(raw_order, ExchangeConstantsOrderColumns.FEE.value, None),
+            "timestamp": get_value_or_default(raw_order, ExchangeConstantsOrderColumns.TIMESTAMP.value, None)
         })
 
-    def __update_type_from_raw(self, raw_order):
-        self.side: TradeOrderSide = TradeOrderSide(raw_order[ExchangeConstantsOrderColumns.SIDE.value])
+    @staticmethod
+    def parse_order_type(raw_order):
+        side: TradeOrderSide = TradeOrderSide(raw_order[ExchangeConstantsOrderColumns.SIDE.value])
         order_type: TradeOrderType = TradeOrderType(raw_order[ExchangeConstantsOrderColumns.TYPE.value])
-        if self.side == TradeOrderSide.BUY:
+
+        if side == TradeOrderSide.BUY:
             if order_type == TradeOrderType.LIMIT:
-                self.order_type = TraderOrderType.BUY_LIMIT
+                order_type = TraderOrderType.BUY_LIMIT
             elif order_type == TradeOrderType.MARKET:
-                self.order_type = TraderOrderType.BUY_MARKET
-        elif self.side == TradeOrderSide.SELL:
+                order_type = TraderOrderType.BUY_MARKET
+        elif side == TradeOrderSide.SELL:
             if order_type == TradeOrderType.LIMIT:
-                self.order_type = TraderOrderType.SELL_LIMIT
+                order_type = TraderOrderType.SELL_LIMIT
             elif order_type == TradeOrderType.MARKET:
-                self.order_type = TraderOrderType.SELL_MARKET
+                order_type = TraderOrderType.SELL_MARKET
+        return side, order_type
+
+    @staticmethod
+    def parse_order_status(raw_order):
+        try:
+            return OrderStatus(raw_order[ExchangeConstantsOrderColumns.STATUS.value])
+        except KeyError:
+            return None
```

Same as above, it a status is not recognized, I think it's better not to silently pass

---

I agree (should raise something) :+1: